debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -557,8 +557,12 @@ protected void handleEvent(Event event) {
         // Update the source offset info. Note that the client returns the value in *milliseconds*, even though the binlog
         // contains only *seconds* precision ...
         EventHeader eventHeader = event.getHeader();
-        source.setBinlogTimestampSeconds(eventHeader.getTimestamp() / 1000L); // client returns milliseconds, but only second
-                                                                              // precision
+        if (!eventHeader.getEventType().equals(EventType.HEARTBEAT)) {
+            //HEARTBEAT events have no timestamp; only set the timestamp if the event is not a HEARTBEAT
```

Comment:
I believe the MySQL build is failing because this change is needed for formatting:
```suggestion
            // HEARTBEAT events have no timestamp; only set the timestamp if the event is not a HEARTBEAT
```